src/com/carolinarollergirls/scoreboard/core/Comparators.java

```diff
@@ -0,0 +1,63 @@
+package com.carolinarollergirls.scoreboard.core;
+
+import java.util.Comparator;
+
+import com.carolinarollergirls.scoreboard.event.NumberedScoreBoardEventProvider;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEventProvider;
+
+public class Comparators {
+    public static class SbepComparator<T extends ScoreBoardEventProvider> implements Comparator<T> {
+        public int compare(ScoreBoardEventProvider p1, ScoreBoardEventProvider p2) {
+            if (p1 == null && p2 == null) { return 0; }
+            if (p2 == null) { return -1; }
+            if (p1 == null) { return 1; }
+            if (p1 instanceof NumberedScoreBoardEventProvider<?> && p2 instanceof NumberedScoreBoardEventProvider<?> &&
+                    p1.getParent() == p2.getParent()) {
+                int n1 = ((NumberedScoreBoardEventProvider<?>)p1).getNumber();
+                int n2 = ((NumberedScoreBoardEventProvider<?>)p2).getNumber();
+                return n1 - n2;
+            } else {
+                return compare(p1.getParent(), p2.getParent());
+            }
+        }
+    }
+
+    public static Comparator<Period> PeriodComparator = new SbepComparator<Period>();
+    public static Comparator<Jam> JamComparator = new SbepComparator<Jam>();
+    public static Comparator<TeamJam> TeamJamComparator = new SbepComparator<TeamJam>();
+    public static Comparator<ScoringTrip> ScoringTripComparator = new SbepComparator<ScoringTrip>();
+    public static Comparator<Fielding> FieldingComparator = new SbepComparator<Fielding>();
+    public static Comparator<BoxTrip> BoxTripComparator = new Comparator<BoxTrip>() {
```

Comment:
Digging a bit more, this would better live in the BoxTrip class implementing the Comparable interface: https://docs.oracle.com/javase/7/docs/api/java/lang/Comparable.html

List.sort will use that automatically then.